src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrScanner.cs

```diff
@@ -37,54 +37,39 @@ public class VmrScanner : IVmrScanner
 
-    public async Task<List<string>> ListCloakedFiles(CancellationToken cancellationToken)
+    public async Task<List<string>> ScanVmr(CancellationToken cancellationToken)
     {
         await _dependencyTracker
             .InitializeSourceMappings(_vmrInfo.VmrPath / VmrInfo.SourcesDir / VmrInfo.SourceMappingsFileName);
 
-        var cloakedFiles = new List<string>();
-
-        foreach (var sourceMapping in _dependencyTracker.Mappings)
-        {
-            cancellationToken.ThrowIfCancellationRequested();
-
-            cloakedFiles.AddRange(await ScanRepository(sourceMapping, cancellationToken));
-        }
-
-        return cloakedFiles;
-    }
-
-    private async Task<string[]> ScanRepository(SourceMapping sourceMapping, CancellationToken cancellationToken)
-    {
-        _logger.LogInformation("Scanning {repository} repository", sourceMapping.Name);
-        var args = new List<string>
-        {
-            "diff",
-            "--name-only",
-            Constants.EmptyGitObject
-        };
+        var taskList = new List<Task<IEnumerable<string>>>();
 
-        var baseExcludePath = _vmrInfo.GetRepoSourcesPath(sourceMapping);
+        _logger.LogInformation("Scanning VMR repositories for {type} files", ScanType());
 
-        foreach (var exclude in sourceMapping.Exclude)
+        foreach (var sourceMapping in _dependencyTracker.Mappings)
         {
-            args.Add(ExcludeFile(baseExcludePath / exclude));
+            taskList.Add(ScanRepository(sourceMapping, cancellationToken));
         }
 
-        var ret = await _processManager.ExecuteGit(_vmrInfo.VmrPath, args.ToArray(), cancellationToken);
+        await Task.WhenAll(taskList);
 
-        ret.ThrowIfFailed($"Failed to scan the {sourceMapping.Name} repository");
-        var files = ret.StandardOutput
-            .Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries);
+        var cloakedFiles = taskList.SelectMany(task => task.Result).OrderBy(file => file);
 
-        foreach (var file in files)
+        if (cloakedFiles.Any())
         {
-            _logger.LogWarning("File {file} is cloaked but present in the VMR", file);
+            _logger.LogInformation("The scanner found {number} {type} files:", cloakedFiles.Count(), ScanType());
+            foreach (var file in cloakedFiles)
+            {
+                _logger.LogWarning("File {file} is {type} but present in the VMR", file, ScanType());
+            }
         }
 
-        return files;
+        return taskList.SelectMany(task => task.Result).ToList();
     }
 
-    private static string ExcludeFile(string file) => $":(attr:!{VmrInfo.KeepAttribute}){file}";
+    protected abstract string ScanType();
+    protected abstract Task<IEnumerable<string>> ScanRepository(SourceMapping sourceMapping, CancellationToken cancellationToken);
+    protected abstract string GetExclusionAttribute();
+    protected string GetExclusionFilter(string file) => $":(attr:!{GetExclusionAttribute()}){file}";
```

Comment:
Why are these here and protected? This class doesn't seem to use them?

Reply:
I tried being fancy and having 1 less line of code. I guess it doesn't really make sense. Made these private in the implementation classes